src/main/java/christmas/model/order/Orders.java

```diff
@@ -0,0 +1,63 @@
+package christmas.model.order;
+
+import christmas.model.ErrorMessage;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Orders {
+
+    private static final int UPPER_LIMIT_OF_TOTAL_MENU_QUANTITY = 20;
+    private final List<Order> orderSheet;
+
+    private Orders(List<Order> orderSheet) {
+        this.orderSheet = orderSheet;
+    }
+
+    public static Orders of(List<Order> orderSheet) {
+        int totalMenuQuantity = orderSheet.stream()
+                .mapToInt(Order::getQuantity)
+                .sum();
+        validateMaxMenuCount(totalMenuQuantity);
+        validateAllDrink(orderSheet);
+        return new Orders(orderSheet);
+    }
+
+    public int calculateTotalOrderCost() {
+        return orderSheet.stream()
+                .mapToInt(Order::calculateOrderCost)
+                .sum();
+    }
+
+    private static void validateMaxMenuCount(int totalMenuQuantity) {
+        if (totalMenuQuantity > UPPER_LIMIT_OF_TOTAL_MENU_QUANTITY) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_ORDER_INPUT.getMessage());
+        }
+    }
+
+    private static void validateAllDrink(List<Order> orderSheet) {
+        boolean isAllDrink = orderSheet.stream()
+                .allMatch(Order::isDrink);
+        if (isAllDrink) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_ORDER_INPUT.getMessage());
+        }
+    }
+
+    public int countDessert() {
+        return orderSheet.stream()
+                .mapToInt(Order::countDessert)
+                .sum();
+    }
+
+    public int countMainMenu() {
+        return orderSheet.stream()
+                .mapToInt(Order::countMainMenu)
+                .sum();
+    }
+
+    public List<OrderDetailDto> createOrderDetailDtos() {
+        return orderSheet.stream()
+                .map(Order::getOrderDetailDto)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
        return orderSheet.stream()
                .map(Order::getOrderDetailDto)
                .toList();
```
해당 부분은 불변성을 위해 `toList()` 사용 권장드립니다!
[참고자료](https://voidmelody.tistory.com/186)입니다.

Reply:
몰랐던 부분이네요 자료 감사합니다!
프리코스 때 생각치도 못한 불변 리스트 에러로 고생했던 기억이... 😂